diff --git a/src/base/threading/platform_thread_freebsd.cc b/src/base/threading/platform_thread_freebsd.cc
index bce1ccf..0b35e5d 100644
--- a/src/base/threading/platform_thread_freebsd.cc
+++ b/src/base/threading/platform_thread_freebsd.cc
@@ -12,7 +12,9 @@
 #include "base/logging.h"
 #include "base/threading/platform_thread_internal_posix.h"
 #include "base/threading/thread_id_name_manager.h"
+#if 0
 #include "base/tracked_objects.h"
+#endif
 #include "build/build_config.h"
 
 #if !defined(OS_NACL)
@@ -67,7 +69,9 @@ bool GetCurrentThreadPriorityForPlatform(ThreadPriority* priority) {
 // static
 void PlatformThread::SetName(const std::string& name) {
   ThreadIdNameManager::GetInstance()->SetName(CurrentId(), name);
+#if 0
   tracked_objects::ThreadData::InitializeThreadContext(name);
+#endif
 
 #if !defined(OS_NACL)
   // On FreeBSD we can get the thread names to show up in the debugger by
