diff --git a/src/net/quic/crypto/quic_crypto_client_config.cc b/src/net/quic/crypto/quic_crypto_client_config.cc
index 1e12424..ac4ed05 100644
--- a/src/net/quic/crypto/quic_crypto_client_config.cc
+++ b/src/net/quic/crypto/quic_crypto_client_config.cc
@@ -6,7 +6,9 @@
 
 #include "base/metrics/histogram.h"
 #include "base/metrics/sparse_histogram.h"
+#if 0
 #include "base/profiler/scoped_tracker.h"
+#endif
 #include "base/stl_util.h"
 #include "base/strings/string_util.h"
 #include "net/quic/crypto/cert_compressor.h"
@@ -343,10 +345,12 @@ void QuicCryptoClientConfig::FillInchoateClientHello(
     const CachedState* cached,
     QuicCryptoNegotiatedParameters* out_params,
     CryptoHandshakeMessage* out) const {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientConfig::FillInchoateClientHello"));
+#endif
 
   out->set_tag(kCHLO);
   out->set_minimum_size(kClientHelloMinimumSize);
@@ -406,10 +410,12 @@ QuicErrorCode QuicCryptoClientConfig::FillClientHello(
     QuicCryptoNegotiatedParameters* out_params,
     CryptoHandshakeMessage* out,
     string* error_details) const {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientConfig::FillClientHello"));
+#endif
 
   DCHECK(error_details != nullptr);
 
diff --git a/src/net/quic/quic_crypto_client_stream.cc b/src/net/quic/quic_crypto_client_stream.cc
index 0798449..18b5aae 100644
--- a/src/net/quic/quic_crypto_client_stream.cc
+++ b/src/net/quic/quic_crypto_client_stream.cc
@@ -238,10 +238,12 @@ void QuicCryptoClientStream::DoHandshakeLoop(
 
 void QuicCryptoClientStream::DoInitialize(
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoInitialize"));
+#endif
 
   if (!cached->IsEmpty() && !cached->signature().empty() &&
       server_id_.is_https()) {
@@ -260,10 +262,12 @@ void QuicCryptoClientStream::DoInitialize(
 void QuicCryptoClientStream::DoSendCHLO(
     const CryptoHandshakeMessage* in,
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile1(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoSendCHLO1"));
+#endif
 
   // Send the client hello in plaintext.
   session()->connection()->SetDefaultEncryptionLevel(ENCRYPTION_NONE);
@@ -314,11 +318,12 @@ void QuicCryptoClientStream::DoSendCHLO(
       &crypto_negotiated_params_,
       &out,
       &error_details);
-
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile2(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoSendCHLO2"));
+#endif
 
   if (error != QUIC_NO_ERROR) {
     // Flush the cached config so that, if it's bad, the server has a
@@ -359,10 +364,12 @@ void QuicCryptoClientStream::DoSendCHLO(
 void QuicCryptoClientStream::DoReceiveREJ(
     const CryptoHandshakeMessage* in,
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoReceiveREJ"));
+#endif
 
   // We sent a dummy CHLO because we didn't have enough information to
   // perform a handshake, or we sent a full hello that the server
@@ -402,10 +409,12 @@ void QuicCryptoClientStream::DoReceiveREJ(
 
 QuicAsyncStatus QuicCryptoClientStream::DoVerifyProof(
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoVerifyProof"));
+#endif
 
   ProofVerifier* verifier = crypto_config_->proof_verifier();
   DCHECK(verifier);
@@ -445,10 +454,12 @@ QuicAsyncStatus QuicCryptoClientStream::DoVerifyProof(
 
 void QuicCryptoClientStream::DoVerifyProofComplete(
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoVerifyProofComplete"));
+#endif
 
   if (!verify_ok_) {
     next_state_ = STATE_NONE;
@@ -479,10 +490,12 @@ void QuicCryptoClientStream::DoVerifyProofComplete(
 
 QuicAsyncStatus QuicCryptoClientStream::DoGetChannelID(
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoGetChannelID"));
+#endif
 
   next_state_ = STATE_GET_CHANNEL_ID_COMPLETE;
   channel_id_key_.reset();
@@ -517,10 +530,12 @@ QuicAsyncStatus QuicCryptoClientStream::DoGetChannelID(
 }
 
 void QuicCryptoClientStream::DoGetChannelIDComplete() {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoGetChannelIDComplete"));
+#endif
 
   if (!channel_id_key_.get()) {
     next_state_ = STATE_NONE;
@@ -534,10 +549,12 @@ void QuicCryptoClientStream::DoGetChannelIDComplete() {
 void QuicCryptoClientStream::DoReceiveSHLO(
     const CryptoHandshakeMessage* in,
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoReceiveSHLO"));
+#endif
 
   next_state_ = STATE_NONE;
   // We sent a CHLO that we expected to be accepted and now we're hoping
@@ -609,10 +626,12 @@ void QuicCryptoClientStream::DoReceiveSHLO(
 
 void QuicCryptoClientStream::DoInitializeServerConfigUpdate(
     QuicCryptoClientConfig::CachedState* cached) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 QuicCryptoClientStream::DoInitializeServerConfigUpdate"));
+#endif
 
   bool update_ignored = false;
   if (!server_id_.is_https()) {
diff --git a/src/net/quic/reliable_quic_stream.cc b/src/net/quic/reliable_quic_stream.cc
index 379991e..4f9ce51 100644
--- a/src/net/quic/reliable_quic_stream.cc
+++ b/src/net/quic/reliable_quic_stream.cc
@@ -5,7 +5,9 @@
 #include "net/quic/reliable_quic_stream.h"
 
 #include "base/logging.h"
+#if 0
 #include "base/profiler/scoped_tracker.h"
+#endif
 #include "net/quic/iovector.h"
 #include "net/quic/quic_flow_controller.h"
 #include "net/quic/quic_session.h"
@@ -224,10 +226,12 @@ void ReliableQuicStream::Reset(QuicRstStreamErrorCode error) {
 }
 
 void ReliableQuicStream::CloseConnection(QuicErrorCode error) {
+#if 0
   // TODO(vadimt): Remove ScopedTracker below once crbug.com/422516 is fixed.
   tracked_objects::ScopedTracker tracking_profile(
       FROM_HERE_WITH_EXPLICIT_FUNCTION(
           "422516 ReliableQuicStream::CloseConnection"));
+#endif
 
   session()->connection()->SendConnectionClose(error);
 }
