diff --git a/src/base/metrics/persistent_memory_allocator.cc b/src/base/metrics/persistent_memory_allocator.cc
index dfa408f..0f32ad1 100644
--- a/src/base/metrics/persistent_memory_allocator.cc
+++ b/src/base/metrics/persistent_memory_allocator.cc
@@ -13,9 +13,13 @@
 #include <sys/mman.h>
 #endif
 
+#if 0
 #include "base/files/memory_mapped_file.h"
+#endif
 #include "base/logging.h"
+#if 0
 #include "base/memory/shared_memory.h"
+#endif
 #include "base/metrics/histogram_macros.h"
 
 namespace {
@@ -777,6 +781,7 @@ void LocalPersistentMemoryAllocator::DeallocateLocalMemory(void* memory,
 
 
 //----- SharedPersistentMemoryAllocator ----------------------------------------
+#if 0
 
 SharedPersistentMemoryAllocator::SharedPersistentMemoryAllocator(
     std::unique_ptr<SharedMemory> memory,
@@ -826,5 +831,6 @@ bool FilePersistentMemoryAllocator::IsFileAcceptable(
   return IsMemoryAcceptable(file.data(), file.length(), 0, read_only);
 }
 #endif  // !defined(OS_NACL)
+#endif
 
 }  // namespace base
