diff --git a/src/base/metrics/persistent_histogram_allocator.cc b/src/base/metrics/persistent_histogram_allocator.cc
index c4d8c95..14855df 100644
--- a/src/base/metrics/persistent_histogram_allocator.cc
+++ b/src/base/metrics/persistent_histogram_allocator.cc
@@ -7,9 +7,11 @@
 #include <memory>
 
 #include "base/files/file_path.h"
+#if 0
 #include "base/files/file_util.h"
 #include "base/files/important_file_writer.h"
 #include "base/files/memory_mapped_file.h"
+#endif
 #include "base/lazy_instance.h"
 #include "base/logging.h"
 #include "base/memory/ptr_util.h"
@@ -684,6 +686,7 @@ void GlobalHistogramAllocator::CreateWithLocalMemory(
       MakeUnique<LocalPersistentMemoryAllocator>(size, id, name))));
 }
 
+#if 0
 #if !defined(OS_NACL)
 // static
 void GlobalHistogramAllocator::CreateWithFile(
@@ -749,6 +752,7 @@ void GlobalHistogramAllocator::CreateWithSharedMemoryHandle(
       new GlobalHistogramAllocator(MakeUnique<SharedPersistentMemoryAllocator>(
           std::move(shm), 0, StringPiece(), /*readonly=*/false))));
 }
+#endif
 
 // static
 void GlobalHistogramAllocator::Set(
@@ -813,6 +817,7 @@ const FilePath& GlobalHistogramAllocator::GetPersistentLocation() const {
 }
 
 bool GlobalHistogramAllocator::WriteToPersistentLocation() {
+#if 0
 #if defined(OS_NACL)
   // NACL doesn't support file operations, including ImportantFileWriter.
   NOTREACHED();
@@ -835,6 +840,10 @@ bool GlobalHistogramAllocator::WriteToPersistentLocation() {
 
   return true;
 #endif
+#else
+  // libquic does not support this
+  return false;
+#endif
 }
 
 GlobalHistogramAllocator::GlobalHistogramAllocator(
diff --git a/src/base/metrics/persistent_histogram_allocator.h b/src/base/metrics/persistent_histogram_allocator.h
index ee1fba5..aa712ec 100644
--- a/src/base/metrics/persistent_histogram_allocator.h
+++ b/src/base/metrics/persistent_histogram_allocator.h
@@ -11,7 +11,9 @@
 #include "base/atomicops.h"
 #include "base/base_export.h"
 #include "base/feature_list.h"
+#if 0
 #include "base/memory/shared_memory.h"
+#endif
 #include "base/metrics/histogram_base.h"
 #include "base/metrics/persistent_memory_allocator.h"
 #include "base/strings/string_piece.h"
@@ -401,6 +403,7 @@ class BASE_EXPORT GlobalHistogramAllocator
                              StringPiece name);
 #endif
 
+#if 0
   // Create a global allocator using a block of shared |memory| of the
   // specified |size|. The allocator takes ownership of the shared memory
   // and releases it upon destruction, though the memory will continue to
@@ -416,6 +419,7 @@ class BASE_EXPORT GlobalHistogramAllocator
   // continue to live if other processes have access to it.
   static void CreateWithSharedMemoryHandle(const SharedMemoryHandle& handle,
                                            size_t size);
+#endif
 
   // Sets a GlobalHistogramAllocator for globally storing histograms in
   // a space that can be persisted or shared between processes. There is only
