diff --git a/src/base/rand_util_posix.cc b/src/base/rand_util_posix.cc
index 6a6e05a..129330a 100644
--- a/src/base/rand_util_posix.cc
+++ b/src/base/rand_util_posix.cc
@@ -10,7 +10,11 @@
 #include <stdint.h>
 #include <unistd.h>
 
+#if 0
 #include "base/files/file_util.h"
+#else
+#include "base/posix/eintr_wrapper.h"
+#endif
 #include "base/lazy_instance.h"
 #include "base/logging.h"
 
@@ -47,6 +51,22 @@ uint64_t RandUint64() {
   return number;
 }
 
+// ReadFromFD is originally defined in base/files/file_util.h
+// libquic removed base/files/file_util.h dependency and copied the function
+// here. Use static to avoid linker conflict in other projects using libquic
+// along with chromium sources.
+static bool ReadFromFD(int fd, char* buffer, size_t bytes) {
+  size_t total_read = 0;
+  while (total_read < bytes) {
+    ssize_t bytes_read =
+        HANDLE_EINTR(read(fd, buffer + total_read, bytes - total_read));
+    if (bytes_read <= 0)
+      break;
+    total_read += bytes_read;
+  }
+  return total_read == bytes;
+}
+
 void RandBytes(void* output, size_t output_length) {
   const int urandom_fd = g_urandom_fd.Pointer()->fd();
   const bool success =
