diff --git a/src/net/quic/quic_sent_packet_manager.cc b/src/net/quic/quic_sent_packet_manager.cc
index c1811cf..b71b8ef 100644
--- a/src/net/quic/quic_sent_packet_manager.cc
+++ b/src/net/quic/quic_sent_packet_manager.cc
@@ -664,7 +664,7 @@ bool QuicSentPacketManager::MaybeRetransmitTailLossProbe() {
     MarkForRetransmission(sequence_number, TLP_RETRANSMISSION);
     return true;
   }
-  DLOG(FATAL)
+  DLOG(ERROR)
     << "No retransmittable packets, so RetransmitOldestPacket failed.";
   return false;
 }
