diff --git a/src/net/spdy/spdy_header_block.cc b/src/net/spdy/spdy_header_block.cc
index 444f270..bd152f8 100644
--- a/src/net/spdy/spdy_header_block.cc
+++ b/src/net/spdy/spdy_header_block.cc
@@ -248,6 +248,7 @@ void SpdyHeaderBlock::AppendHeader(const StringPiece key,
   block_.insert(make_pair(storage_->Write(key), storage_->Write(value)));
 }
 
+#if 0
 std::unique_ptr<base::Value> SpdyHeaderBlockNetLogCallback(
     const SpdyHeaderBlock* headers,
     NetLogCaptureMode capture_mode) {
@@ -263,6 +264,7 @@ std::unique_ptr<base::Value> SpdyHeaderBlockNetLogCallback(
   dict->Set("headers", headers_dict);
   return std::move(dict);
 }
+#endif
 
 bool SpdyHeaderBlockFromNetLogParam(
     const base::Value* event_param,
